manager/src/main/java/com/redhat/service/bridge/manager/workers/resources/AbstractWorker.java

```diff
@@ -0,0 +1,137 @@
+                    managedResource.getName(),
+                    managedResource.getId());
+            managedResource.setStatus(ManagedResourceStatus.FAILED);
+            persist(managedResource);
+            return managedResource;
+        }
+        if (isTimeoutExceeded(work)) {
+            LOGGER.error(
+                    "Timeout exceeded trying to create dependencies for '{}' [{}].",
+                    managedResource.getName(),
+                    managedResource.getId());
+            managedResource.setStatus(ManagedResourceStatus.FAILED);
+            persist(managedResource);
+            return managedResource;
+        }
+
+        boolean complete = false;
+        T updated = managedResource;
+        if (PROVISIONING_STARTED.contains(managedResource.getStatus())) {
+            try {
+                updated = createDependencies(work, managedResource);
+                complete = PROVISIONING_COMPLETED.contains(updated.getStatus());
+            } catch (Exception e) {
+                LOGGER.info(
+                        "Failed to create dependencies for '{}' [{}].\n"
+                                + "Work status: {}\n"
+                                + "{}",
+                        managedResource.getName(),
+                        managedResource.getId(),
+                        work,
+                        e.getMessage());
+                // Something has gone wrong. We need to retry.
+                workManager.recordAttempt(work);
+            }
+
+        } else if (DEPROVISIONING_STARTED.contains(managedResource.getStatus())) {
+            try {
+                updated = deleteDependencies(work, managedResource);
+                complete = DEPROVISIONING_COMPLETED.contains(updated.getStatus());
+            } catch (Exception e) {
+                LOGGER.info("Failed to delete dependencies for '{}' [{}].\n"
+                        + "Work status: {}\n"
+                        + "{}",
+                        managedResource.getName(),
+                        managedResource.getId(),
+                        work,
+                        e.getMessage());
+                // Something has gone wrong. We need to retry.
+                workManager.recordAttempt(work);
+            }
+        }
+
+        if (complete) {
+            workManager.complete(work);
+        }
+
+        return updated;
+    }
+
+    @Transactional
+    protected T load(Work work) {
+        return getDao().findById(work.getManagedResourceId());
+    }
+
+    @Transactional
+    protected T persist(T managedResource) {
+        return getDao().getEntityManager().merge(managedResource);
+    }
+
+    protected boolean areRetriesExceeded(Work w) {
+        return w.getAttempts() > maxRetries;
+    }
+
+    protected boolean isTimeoutExceeded(Work work) {
+        return ZonedDateTime.now().minusSeconds(timeoutSeconds).isAfter(work.getSubmittedAt());
+    }
+
```

Comment:
and actually we record a lot of attempts. We might exaust the number of attempts in few seconds maybe?